lxd/lifecycle/lifecycle.go

```diff
@@ -0,0 +1,17 @@
+package lifecycle
+
+type InstanceAction string
+
+const (
+	CREATED   = InstanceAction("created")
+	STARTED   = InstanceAction("started")
+	STOPPED   = InstanceAction("stopped")
+	SHUTDOWN  = InstanceAction("shutdown")
+	RESTARTED = InstanceAction("restarted")
+	PAUSED    = InstanceAction("paused")
+	RESUMED   = InstanceAction("resumed")
+	RESTORED  = InstanceAction("restored")
+	DELETED   = InstanceAction("deleted")
+	RENAMED   = InstanceAction("renamed")
+	UPDATED   = InstanceAction("updated")
```

Comment:
I expect that `golint` won't like the all caps variables and will want a description string.

So we'll probably want things like `InstanceCreated`.
For dealing with the issue of the different strings for instances and instance snapshots, I wonder if we can't do something like:


```
type InstanceAction string
func (a InstanceAction) Instance() {
    return fmt.Sprintf("instance-%s", a)
}

func (a InstanceAction) Snapshot() {
    return fmt.Sprintf("instance-snapshot-%s", a)
}

const InstanceDeleted = InstanceAction("deleted")
```

So that from the driver code, we can call .Instance() or .Snapshot() on it depending on which we want?